tornado/httputil.py

```diff
@@ -243,6 +244,86 @@ def __deepcopy__(self, memo_dict):
         return self.copy()
 
 
+_cookie_value_decode = Cookie.SimpleCookie().value_decode
+_cookie_attr_reserved = Cookie.Morsel._reserved
+_cookie_attr_flags = set(["secure", "httponly"])
+
+# part of Cookie library module
+# https://docs.python.org/2.7/library/cookie.html
```

Comment:
So the only change here is the try/except in the parse_cookies function? That's all I've been asking; what is copied and what is changed. This seems like a lot of duplication just to add that. (and you're still depending on private details of the Cookie module, like `Cookie.Morsel._reserved`)

Why does it matter how common invalid cookies are "in the real world" when you only have to deal with cookies on your own domain (i.e. ones you set)? 

I'm concerned about the security implications of a change like this. Could differences in lenient cookie parsing allow someone to send a cookie header that would be parsed differently by different components? It concerns me that by catching exceptions instead of changing the regular expression, it is difficult to say precisely how this change deviates from the spec and where parsing will resume after an error.

By contrast, http://wiki.apache.org/tomcat/Cookies specifies a "lenient RFC6256 parser" [sic]: names may contain any ascii char but semicolon or equals; values may contain any utf8 but semicolons. 

It looks like even RFC 6265 is doomed; Chrome, IE, and Safari ignore double quotes and split on semicolons: http://neugierig.org/software/chromium/notes/2009/01/cookies.html